### PR TITLE
Handle no-content response from clickUrl

### LINKED ISSUE #11790: Expect 204 in response to clickUrl when ReplaceUrl is present
When clickUrl is present, the expected response today is a JSON containing the final location, which  is used to propagate tracking parameters to the runtime. When replaceUrl is present, tracking params will come solely from the replaceUrl, and clickUrl will return a 204 - this needs to be handled.

### PR DESCRIPTION
**Pull request by zhouyx**
close #11790 

Please note that 204 response is only expected when `replaceUrl` is present. However I feel even without `replaceUrl`, a no-content response should still be handled without throwing an error/warning. Let me know if anyone thinks that I should check the existence of `replaceUrl` specifically. 

cc @sucharithav

### CODE REVIEW

File: src/impression.js [hunk 1]
```diff
@@ -187,16 +190,26 @@ function invoke(win, clickUrl) {
     credentials: 'include',
```
Comment (sucharithav) [inline src/impression.js]: Wondering if it's appropriate to continue to use "fetchJson" here?

Comment (Ildave012222) [reply src/impression.js]: Hi

Comment (zhouyx) [reply src/impression.js]: I think `#fetchJson` is ok to use here. 
`#fetchJson` merely sets up the request to accept JSON. For the special no-content case, we resolve with null response with status 204 anyway. The real `response.json()` is not performed.

Comment (sucharithav) [reply src/impression.js]: Sounds good, thanks!

**Pull request merged**